src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/custom.py

```diff
@@ -0,0 +1,167 @@
+    :return:
+    """
+    from azure.mgmt.devtestlabs.models.lab_virtual_machine import LabVirtualMachine
+
+    is_authentication_with_ssh_key = True if authentication_type == 'ssh' else False
+
+    lab_virtual_machine = LabVirtualMachine(notes=notes,
+                                            custom_image_id=custom_image_id,
+                                            size=size,
+                                            user_name=admin_username,
+                                            password=admin_password,
+                                            ssh_key=ssh_key,
+                                            is_authentication_with_ssh_key=is_authentication_with_ssh_key,
+                                            lab_subnet_name=subnet,
+                                            lab_virtual_network_id=lab_virtual_network_id,
+                                            disallow_public_ip_address=disallow_public_ip_address,
+                                            artifacts=artifacts,
+                                            gallery_image_reference=gallery_image_reference,
+                                            name=name,
+                                            location=location,
+                                            tags=tags)
+
+    return client.create_environment(resource_group, lab_name, lab_virtual_machine)
+
+
+def list_vm(client, lab_name, resource_group, order_by=None, top=None,
+            filters=None, my_vms=None, my_claimable_vms=None):
+    """
+    :param client: Virtual machine client
+    :param lab_name: Name of the lab
+    :param resource_group: Name of the lab resource group
+    :param order_by: The ordering expression for the results, using OData notation
+    :param top: The maximum number of resources to return from the operation
+    :param filters: The filter to apply on the operation
+    :param my_vms: List of viewable labs virtual machines. Cannot be used with --filters
+    :param my_claimable_vms: List of claimable labs virtual machines. Cannot be used with --filters
+    :return:
+    """
+
+    collection = [filters, my_vms, my_claimable_vms]
+    if not _single(collection):
+        raise CLIError("usage error: [--filters FILTER | --my-vms | --my-claimable-vms]")
+
+    if my_vms:
+        # Find out owner object id
+        from azure.cli.core._profile import Profile, CLOUD
+        profile = Profile()
+        cred, _, tenant_id = profile.get_login_credentials(
+            resource=CLOUD.endpoints.active_directory_graph_resource_id)
+        graph_client = GraphRbacManagementClient(cred,
+                                                 tenant_id,
+                                                 base_url=CLOUD.endpoints.active_directory_graph_resource_id)
+        subscription = profile.get_subscription()
+        object_id = _get_object_id(graph_client, subscription=subscription)
+        odata_filter = "Properties/ownerObjectId eq '{}'".format(object_id)
```

Comment:
I did confirm with the DTL team and they REST endpoint will work with service principal as well.